build-system/babel-plugins/babel-plugin-amp-mode-transformer/index.js

```diff
@@ -59,15 +51,19 @@ module.exports = function ({types: t}) {
         const {node} = path;
         const {object: obj, property} = node;
         const {callee} = obj;
+        const {INTERNAL_RUNTIME_VERSION, IS_ESM} = this.opts;
+
         if (callee && callee.name === 'getMode') {
           if (property.name === 'test' || property.name === 'localDev') {
             path.replaceWith(t.booleanLiteral(false));
-          }
-          if (shouldResolveDevelopmentMode && property.name === 'development') {
+          } else if (property.name === 'development') {
             path.replaceWith(t.booleanLiteral(false));
-          }
-          if (property.name === 'minified') {
+          } else if (property.name === 'minified') {
             path.replaceWith(t.booleanLiteral(true));
+          } else if (property.name === 'esm') {
+            path.replaceWith(t.booleanLiteral(IS_ESM));
+          } else if (property.name === 'version') {
+            path.replaceWith(t.stringLiteral(INTERNAL_RUNTIME_VERSION));
```

Comment:
```javascript
if (callee?.name == 'getMode') {
  const {name} = property;
  const flags = {
    test: false,
    localDev: false,
    development: false,
    minified: true,
    esm: IS_ESM,
  };
  if (name in flags) {
    path.replaceWith(t.booleanLiteral(flags[name]));
  } else if (name == 'version') {
    path.replaceWith(t.stringLiteral(INTERNAL_RUNTIME_VERSION));
  }
}
```

Reply:
```js
if (callee?.name == 'getMode') {
  const {name} = property;
  const flags = {
    test: false,
    localDev: false,
    development: false,
    minified: true,
    esm: IS_ESM,
  version: INTERNAL_RUNTIME_VERSION
  };
  if (name in flags) {
    path.replaceWith(t.booleanLiteral(flags[name]));
  }
}
```

Reply:
I'll pursue this cleanup in a followup PR, after making this transform also work on the `mode` namespace object

Reply:
I don't think we need to be transforming these booleans at all; the compiler will already handle this with DCE

Reply:
Agree. Can test this out in a followup!

Reply:
It doesn't feel right to be adding new code to transform something that we likely agree doesn't need to be transformed?